exporter/tanzuobservabilityexporter/metrics.go

```diff
@@ -263,3 +270,195 @@ func (g *gaugeConsumer) pushSingleNumberDataPoint(
+		sender:                sender,
+		reporting:             newHistogramReporting(fixedOptions.Logger),
+		reportInternalMetrics: fixedOptions.ReportInternalMetrics,
+	}
+}
+
+func (h *histogramConsumer) Type() pdata.MetricDataType {
+	return pdata.MetricDataTypeHistogram
+}
+
+func (h *histogramConsumer) Consume(metric pdata.Metric, errs *[]error) {
+	histogram := metric.Histogram()
+	aggregationTemporality := histogram.AggregationTemporality()
+	var consumer histogramDataPointConsumer
+	switch aggregationTemporality {
+	case pdata.MetricAggregationTemporalityDelta:
+		// TODO: follow-on pull request will add support for Delta Histograms.
+		*errs = append(*errs, errors.New("delta histograms not supported"))
+		return
+	case pdata.MetricAggregationTemporalityCumulative:
+		consumer = h.cumulative
+	default:
+		h.reporting.LogNoAggregationTemporality(metric)
+		return
+	}
+	histogramDataPoints := histogram.DataPoints()
+	for i := 0; i < histogramDataPoints.Len(); i++ {
+		consumer.Consume(metric, histogramDataPoints.At(i), errs, h.reporting)
+	}
+}
+
+func (h *histogramConsumer) PushInternalMetrics(errs *[]error) {
+	if h.reportInternalMetrics {
+		h.reporting.Report(h.sender, errs)
+	}
+}
+
+// histogramDataPointConsumer consumes one histogram data point. There is one
+// implementation for delta histograms and one for cumulative histograms.
+type histogramDataPointConsumer interface {
+
+	// Consume consumes the histogram data point.
+	// metric is the enclosing metric; histogram is the histogram data point;
+	// errors get appended to errs; reporting keeps track of special situations
+	Consume(
+		metric pdata.Metric,
+		histogram pdata.HistogramDataPoint,
+		errs *[]error,
+		reporting *histogramReporting,
+	)
+}
+
+type cumulativeHistogramDataPointConsumer struct {
+	sender gaugeSender
+}
+
+// newCumulativeHistogramDataPointConsumer returns a consumer for cumulative
+// histogram data points.
+func newCumulativeHistogramDataPointConsumer(sender gaugeSender) histogramDataPointConsumer {
+	return &cumulativeHistogramDataPointConsumer{sender: sender}
+}
+
+func (c *cumulativeHistogramDataPointConsumer) Consume(
+	metric pdata.Metric,
+	h pdata.HistogramDataPoint,
+	errs *[]error,
+	reporting *histogramReporting,
+) {
+	name := metric.Name()
+	tags := attributesToTags(h.Attributes())
+	ts := h.Timestamp().AsTime().Unix()
+	explicitBounds := h.ExplicitBounds()
+	bucketCounts := h.BucketCounts()
+	var abort bool
+	if len(bucketCounts) != len(explicitBounds)+1 {
+		reporting.LogMalformed(metric)
+		abort = true
+	}
+	if _, ok := tags["le"]; ok {
+		reporting.LogLeTagInUse(metric)
+		abort = true
+	}
+	if abort {
+		return
+	}
+	var leCount uint64
+	for i := range bucketCounts {
+		tags["le"] = leTagValue(explicitBounds, i)
+		leCount += bucketCounts[i]
+		err := c.sender.SendMetric(name, float64(leCount), ts, "", tags)
+		if err != nil {
+			*errs = append(*errs, err)
+		}
+	}
+}
+
+func leTagValue(explicitBounds []float64, idx int) string {
```

Comment:
Would it make sense to rename `idx` to `bucketIndex`?

Reply:
Done.